homeassistant/components/heatzy/climate.py

```diff
@@ -0,0 +1,188 @@
+
+
+class HeatzyThermostat(CoordinatorEntity, ClimateEntity):
+    """Heatzy climate."""
+
+    def __init__(self, coordinator, unique_id):
+        """Init."""
+        super().__init__(coordinator)
+        self._coordinator = coordinator
+        self._unique_id = unique_id
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_PRESET_MODE
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._unique_id
+
+    @property
+    def name(self):
+        """Return a name."""
+        return self._coordinator.data[self.unique_id]["dev_alias"]
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            "name": self.name,
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "manufacturer": DOMAIN,
+            "model": self._coordinator.data[self.unique_id].get("product_name"),
+            "sw_version": self._coordinator.data[self.unique_id].get(
+                "wifi_soft_version"
+            ),
+        }
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available hvac operation modes."""
+        return MODE_LIST
+
+    @property
+    def hvac_mode(self):
+        """Return hvac operation ie. heat, cool mode."""
+        if self.preset_mode == PRESET_NONE:
+            return HVAC_MODE_OFF
+        return HVAC_MODE_HEAT
+
+    @property
+    def preset_modes(self):
+        """Return a list of available preset modes."""
+        return PRESET_LIST
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set new hvac mode."""
+        if hvac_mode == HVAC_MODE_OFF:
+            await self.async_turn_off()
+        elif hvac_mode == HVAC_MODE_HEAT:
+            await self.async_turn_on()
+
+    async def async_turn_on(self):
+        """Turn device on."""
+        await self.async_set_preset_mode(PRESET_COMFORT)
+
+    async def async_turn_off(self):
+        """Turn device off."""
+        await self.async_set_preset_mode(PRESET_NONE)
+
+
+class HeatzyPiloteV1Thermostat(HeatzyThermostat):
+    """Heaty Pilote v1."""
+
+    HEATZY_TO_HA_STATE = {
+        "\u8212\u9002": PRESET_COMFORT,
+        "\u7ecf\u6d4e": PRESET_ECO,
+        "\u89e3\u51bb": PRESET_AWAY,
+        "\u505c\u6b62": PRESET_NONE,
+    }
+    HA_TO_HEATZY_STATE = {
+        PRESET_COMFORT: [1, 1, 0],
+        PRESET_ECO: [1, 1, 1],
+        PRESET_AWAY: [1, 1, 2],
+        PRESET_NONE: [1, 1, 3],
+    }
+
+    @property
+    def preset_mode(self):
+        """Return the current preset mode, e.g., home, away, temp."""
+        return self.HEATZY_TO_HA_STATE.get(
+            self._coordinator.data[self.unique_id].get("attr", {}).get("mode")
+        )
+
+    async def async_set_preset_mode(self, preset_mode):
```

Comment:
The API is sync, so we can just use the sync API Home assistant offers here:

```suggestion
    def set_preset_mode(self, preset_mode):
```

And adjust the method accordingly (removing add_executor_job)